# Churn case. Part 2. Analysis of LTV

Lifetime value (LTV) is the total worth to a business of a customer over the while period of their relationship. It's an important metric as it costs less to keep existings customers than to acquire new ones, so the idea of increasing the value of your existig customers is a great way to drive growth. 

In [1]:
import sqlite3
import pandas as pd

In [2]:
conn = sqlite3.connect("churn.db")

In [3]:
df = pd.read_sql("select * from churn_all", conn)

In [4]:
df.head()

CustomerID  Gender SeniorCitizen Partner Dependents       State   Latitude  \
0  3668-QPYBK    Male            No      No         No  California  33.964131   
1  9237-HQITU  Female            No      No        Yes  California  34.059281   
2  9305-CDSKC  Female            No      No        Yes  California  34.048013   
3  7892-POOKP  Female            No     Yes        Yes  California  34.062125   
4  0280-XJGEX    Male            No      No        Yes  California  34.039224   

    Longitude  ZipCode PhoneService  ... TechSupport StreamingTV  \
0 -118.272783    90003          Yes  ...          No          No   
1 -118.307420    90005          Yes  ...          No          No   
2 -118.293953    90006          Yes  ...          No         Yes   
3 -118.315709    90010          Yes  ...         Yes         Yes   
4 -118.266293    90015          Yes  ...          No         Yes   

  StreamingMovies Tenure        Contract              PaymentMethod  \
0              No      2  Month-to-month               Mailed check   
1              No      2  Month-to-month           Electronic check   
2             Yes      8  Month-to-month           Electronic check   
3             Yes     28  Month-to-month           Electronic check   
4             Yes     49  Month-to-month  Bank transfer (automatic)   

  PaperlessBilling MonthlyCharges  TotalCharges Churn  
0              Yes          53.85        108.15   Yes  
1              Yes          70.70        151.65   Yes  
2              Yes          99.65        820.50   Yes  
3              Yes         104.80       3046.05   Yes  
4              Yes         103.70       5036.30   Yes  

[5 rows x 25 columns]

## 1. For those customers who unsubscribed the service, what was their average LTV? and how long did they usually stay in the service?

In [5]:
## Extract those who unsubscribed the serviced (Churn = Yes)

churn_df = pd.read_sql("select * from churn_all where Churn = 'Yes'", conn)

In [6]:
# Examine the distribution of TotalCharges
churn_df["TotalCharges"].describe()

count    1869.000000
mean     1531.796094
std      1890.822994
min        18.850000
25%       134.500000
50%       703.550000
75%      2331.300000
max      8684.800000
Name: TotalCharges, dtype: float64

We can see that around 20% of the TotalCharges are very high, so let's divide the data to see each distribution.

In [7]:
# Find the 80th percentile of the data in TotalCharges
churn_df.TotalCharges.quantile(0.8)

2840.4100000000003

In [9]:
pd.read_sql("select * from churn_all where TotalCharges <= 2840.41", conn)

CustomerID  Gender SeniorCitizen Partner Dependents       State  \
0     3668-QPYBK    Male            No      No         No  California   
1     9237-HQITU  Female            No      No        Yes  California   
2     9305-CDSKC  Female            No      No        Yes  California   
3     4190-MFLUW  Female            No     Yes         No  California   
4     8779-QRDMV    Male           Yes      No         No  California   
5     1066-JKSGK    Male            No      No         No  California   
6     8665-UTDHZ    Male            No     Yes         No  California   
7     8773-HHUOZ  Female            No      No         No  California   
8     6047-YHPVI    Male            No      No        Yes  California   
9     8168-UQWWF  Female            No      No        Yes  California   
10    7760-OYPDY  Female            No      No        Yes  California   
11    9420-LOJKX  Female            No      No        Yes  California   
12    7495-OOKFY  Female           Yes     Yes        Yes  California   
13    1658-BYGOY    Male           Yes      No        Yes  California   
14    5698-BQJOH  Female            No      No        Yes  California   
15    5919-TMRGD  Female            No      No        Yes  California   
16    9191-MYQKX  Female            No     Yes        Yes  California   
17    8637-XJIVR  Female            No      No        Yes  California   
18    0278-YXOOG    Male            No      No         No  California   
19    4598-XLKNJ  Female           Yes     Yes        Yes  California   
20    5299-RULOA  Female            No     Yes        Yes  California   
21    0404-SWRVG    Male            No      No        Yes  California   
22    4412-YLTKF  Female           Yes      No        Yes  California   
23    6207-WIOLX  Female            No     Yes         No  California   
24    3091-FYHKI    Male            No      No         No  California   
25    2372-HWUHI    Male            No      No         No  California   
26    0390-DCFDQ  Female           Yes     Yes        Yes  California   
27    4080-OGPJL  Female            No      No         No  California   
28    2135-RXIHG    Male            No      No         No  California   
29    3874-EQOEP    Male            No      No        Yes  California   
...          ...     ...           ...     ...        ...         ...   
4709  6598-KELSS    Male            No      No        Yes  California   
4710  1904-WAJAA  Female            No     Yes        Yes  California   
4711  2843-CQMEG    Male            No      No         No  California   
4712  5351-QESIO    Male            No      No        Yes  California   
4713  5568-DMXZS  Female            No      No         No  California   
4714  6633-SYEUS  Female            No      No         No  California   
4715  6447-GORXK    Male            No      No        Yes  California   
4716  6967-PEJLL    Male            No     Yes        Yes  California   
4717  1684-FLBGS  Female            No     Yes        Yes  California   
4718  1389-WNUIB  Female            No     Yes        Yes  California   
4719  0804-YGEQV  Female            No     Yes        Yes  California   
4720  2523-EWWZL  Female            No     Yes         No  California   
4721  0928-XUTSN  Female            No      No         No  California   
4722  0052-YNYOT  Female            No      No         No  California   
4723  4501-VCPFK    Male            No      No         No  California   
4724  9347-AERRL    Male            No     Yes         No  California   
4725  1471-GIQKQ  Female            No      No         No  California   
4726  4807-IZYOZ  Female            No      No         No  California   
4727  9710-NJERN  Female            No      No         No  California   
4728  9837-FWLCH    Male            No     Yes        Yes  California   
4729  2823-LKABH  Female            No      No         No  California   
4730  0550-DCXLH    Male            No      No         No  California   
4731  2235-DWLJU  Female           Yes      No         No  Calif

In [10]:
churn_df.query("TotalCharges <= 2840.41")

CustomerID  Gender SeniorCitizen Partner Dependents       State  \
0     3668-QPYBK    Male            No      No         No  California   
1     9237-HQITU  Female            No      No        Yes  California   
2     9305-CDSKC  Female            No      No        Yes  California   
5     4190-MFLUW  Female            No     Yes         No  California   
6     8779-QRDMV    Male           Yes      No         No  California   
7     1066-JKSGK    Male            No      No         No  California   
9     8665-UTDHZ    Male            No     Yes         No  California   
10    8773-HHUOZ  Female            No      No         No  California   
11    6047-YHPVI    Male            No      No        Yes  California   
13    8168-UQWWF  Female            No      No        Yes  California   
14    7760-OYPDY  Female            No      No        Yes  California   
15    9420-LOJKX  Female            No      No        Yes  California   
16    7495-OOKFY  Female           Yes     Yes        Yes  California   
17    1658-BYGOY    Male           Yes      No        Yes  California   
18    5698-BQJOH  Female            No      No        Yes  California   
19    5919-TMRGD  Female            No      No        Yes  California   
20    9191-MYQKX  Female            No     Yes        Yes  California   
21    8637-XJIVR  Female            No      No        Yes  California   
22    0278-YXOOG    Male            No      No         No  California   
23    4598-XLKNJ  Female           Yes     Yes        Yes  California   
27    5299-RULOA  Female            No     Yes        Yes  California   
28    0404-SWRVG    Male            No      No        Yes  California   
29    4412-YLTKF  Female           Yes      No        Yes  California   
30    6207-WIOLX  Female            No     Yes         No  California   
31    3091-FYHKI    Male            No      No         No  California   
32    2372-HWUHI    Male            No      No         No  California   
33    0390-DCFDQ  Female           Yes     Yes        Yes  California   
34    4080-OGPJL  Female            No      No         No  California   
35    2135-RXIHG    Male            No      No         No  California   
36    3874-EQOEP    Male            No      No        Yes  California   
...          ...     ...           ...     ...        ...         ...   
1829  3926-CUQZX    Male            No      No         No  California   
1831  1750-CSKKM    Male            No      No        Yes  California   
1834  6457-GIRWB    Male            No      No         No  California   
1836  5570-PTWEH  Female            No     Yes         No  California   
1837  6502-MJQAE    Male            No      No         No  California   
1838  7693-LCKZL    Male            No     Yes        Yes  California   
1839  0032-PGELS  Female            No     Yes        Yes  California   
1842  1564-NTYXF  Female           Yes      No         No  California   
1843  8155-IBNHG  Female            No     Yes         No  California   
1844  0886-QGENL  Female           Yes     Yes         No  California   
1845  8739-WWKDU    Male           Yes      No         No  California   
1846  8685-WHQPW  Female           Yes      No         No  California   
1847  4745-LSPLO    Male            No      No         No  California   
1848  8083-YTZES    Male            No      No         No  California   
1849  7240-FQLHE  Female           Yes     Yes         No  California   
1851  0786-VSSUD    Male           Yes      No         No  California   
1852  3976-BWUCK  Female            No     Yes         No  California   
1853  3585-ISXZP  Female            No      No         No  California   
1855  4174-LPGTI  Female            No     Yes         No  California   
1856  2108-XWMPY    Male            No      No         No  California   
1857  6304-IJFSQ    Male            No      No         No  California   
1859  1980-KXVPM  Female           Yes      No         No  California   
1860  7703-ZEKEF    Male            No      No         No  Calif

In [12]:
# Divide the data by the 80th percentile of the data from the TotalCharges variable
total_charges_under80 = churn_df.query("TotalCharges <= 2840.41")
total_charges_above80 = churn_df.query("TotalCharges > 2840.41")

In [14]:
# Show the distribution of people under the 80th percentile
total_charges_under80.TotalCharges.describe()

count    1495.000000
mean      711.265819
std       766.848197
min        18.850000
25%        85.025000
50%       371.650000
75%      1128.225000
max      2839.650000
Name: TotalCharges, dtype: float64

In [15]:
# Show the distribution of people above the 80th percentile
total_charges_above80.TotalCharges.describe()

count     374.000000
mean     4811.723262
std      1436.724288
min      2841.550000
25%      3522.462500
50%      4571.100000
75%      5891.212500
max      8684.800000
Name: TotalCharges, dtype: float64

In [16]:
# Show the distribution of the tenure of people under the 80th percentile
total_charges_under80.Tenure.describe()

count    1495.000000
mean        9.935117
std        10.742349
min         1.000000
25%         1.000000
50%         6.000000
75%        15.000000
max        61.000000
Name: Tenure, dtype: float64

In [17]:
# Show the distribution of the tenure of people above the 80th percentile
total_charges_above80.Tenure.describe()

count    374.000000
mean      50.133690
std       12.334841
min       27.000000
25%       40.000000
50%       49.500000
75%       60.000000
max       72.000000
Name: Tenure, dtype: float64

The average LTV of the 80% of those who unsubscribed is \\$ 711, and their tenure is near 10 months. On the hand, the average LTV of the 20 % of those who unsubscribed is $ 4811, and their tenure is near 50 months.  

## 2. What kinds of services they subscribed when they were still a customer?

### PhoneService

In [18]:
# under 80
pd.read_sql("select count(1) from churn_all where TotalCharges <= 2840.41 and Churn = 'Yes' group by PhoneService", conn)

count(1)
0       161
1      1334

In [20]:
df["Gender"].value_counts() # example

Male      3549
Female    3483
Name: Gender, dtype: int64

In [21]:
# under 80
total_charges_under80.PhoneService.value_counts()

Yes    1334
No      161
Name: PhoneService, dtype: int64

In [22]:
# Example of shape
df.shape

(7032, 25)

In [23]:
# Under 80 (percentage)
total_charges_under80.PhoneService.value_counts() / total_charges_under80.shape[0]

Yes    0.892308
No     0.107692
Name: PhoneService, dtype: float64

In [24]:
# Above 80 (percentage)
total_charges_above80.PhoneService.value_counts() / total_charges_above80.shape[0]

Yes    0.975936
No     0.024064
Name: PhoneService, dtype: float64

For the top 20% (high LTV) only 2% of them didn't use the phone service. On the other hand, lower 80% (LTV) had around 11% of the people that didn't use the phone service. That is around 5 times more as the top 20%. Apparently if you are not subscribed to the phone service you are more likely to be in the bottom 80% of the LTV. We as a company can try to sell this service to more people.

### MutipleLines

In [25]:
total_charges_under80_use_phone = total_charges_under80.query("PhoneService == 'Yes'")

total_charges_under80_use_phone.MultipleLines.value_counts() / total_charges_under80_use_phone.shape[0]

No     0.590705
Yes    0.409295
Name: MultipleLines, dtype: float64

In [27]:
total_charges_above80_use_phone = total_charges_above80.query("PhoneService == 'Yes'")

total_charges_above80_use_phone.MultipleLines.value_counts() / total_charges_above80_use_phone.shape[0]

Yes    0.832877
No     0.167123
Name: MultipleLines, dtype: float64

For the top 20% of LTV who used the phone service 84% of them used multiple lines, which is 2 times as much as the proportion of people in the low 80% LTV that is around 40%. Having multiple lines is going to get you closer to be in the top 20% of the LTV.

### InternetService

In [33]:
# Under 80 (percentage)
total_charges_under80.InternetService.value_counts() / total_charges_under80.shape[0]

Fiber optic    0.640803
DSL            0.283612
No             0.075585
Name: InternetService, dtype: float64

In [34]:
# Above 80 (percentage)
total_charges_above80.InternetService.value_counts() / total_charges_above80.shape[0]

Fiber optic    0.906417
DSL            0.093583
Name: InternetService, dtype: float64

All the people in the high 20% LTV used internet service. On the other hand 8% of the people in the low 80th percentile of the LTV did not have internet connection. Also, in the top 20% (high LTV) 90% of them had fiber optic, in comparison to only 64% in the low LTV. So as a company if we want to increase the LTV of customers, we should recommend using internet and Fiber Optic connection.

### Other services

In [34]:
import numpy as np

In [35]:
total_charges_under80_use_internet = total_charges_under80.query('InternetService!="No"')
proportion_internet_sub_service_under80 = np.array([total_charges_under80_use_internet.query('OnlineSecurity=="Yes"').shape[0]/total_charges_under80_use_internet.shape[0],
total_charges_under80_use_internet.query('TechSupport=="Yes"').shape[0]/total_charges_under80_use_internet.shape[0],
total_charges_under80_use_internet.query('OnlineBackup=="Yes"').shape[0]/total_charges_under80_use_internet.shape[0],
total_charges_under80_use_internet.query('DeviceProtection=="Yes"').shape[0]/total_charges_under80_use_internet.shape[0],
total_charges_under80_use_internet.query('StreamingTV=="Yes"').shape[0]/total_charges_under80_use_internet.shape[0],
total_charges_under80_use_internet.query('StreamingMovies=="Yes"').shape[0]/total_charges_under80_use_internet.shape[0]])

In [36]:
proportion_internet_sub_service_under80

array([0.1316932 , 0.13603473, 0.22720695, 0.23516643, 0.37698987,
       0.3813314 ])

In [38]:
total_charges_above80_use_internet = total_charges_above80.query('InternetService!="No"')
proportion_internet_sub_service_above80 = np.array([total_charges_above80_use_internet.query('OnlineSecurity=="Yes"').shape[0]/total_charges_above80_use_internet.shape[0],
total_charges_above80_use_internet.query('TechSupport=="Yes"').shape[0]/total_charges_above80_use_internet.shape[0],
total_charges_above80_use_internet.query('OnlineBackup=="Yes"').shape[0]/total_charges_above80_use_internet.shape[0],
total_charges_above80_use_internet.query('DeviceProtection=="Yes"').shape[0]/total_charges_above80_use_internet.shape[0],
total_charges_above80_use_internet.query('StreamingTV=="Yes"').shape[0]/total_charges_above80_use_internet.shape[0],
total_charges_above80_use_internet.query('StreamingMovies=="Yes"').shape[0]/total_charges_above80_use_internet.shape[0]])

In [39]:
proportion_internet_sub_service_above80

array([0.30213904, 0.32620321, 0.55882353, 0.58823529, 0.78342246,
       0.77807487])

In the top 20% LTV, both streaming movies and streaming tv percentages are around 80%, in comparison in the low 80% LTV they are around 40%. Device protection and online backup for the top 20% LTV are around 55%, in comparison to only around 22% in the low LTV. As a company we would like to sell to customers streaming TV and movies and also online backup and device protection if we want to increase LTV.

## 3. For those who churned what is the proportion of each kind of contract?

In [40]:
# Under 80 (percentage)
total_charges_under80.Contract.value_counts() / total_charges_under80.shape[0]

Month-to-month    0.959197
One year          0.036789
Two year          0.004013
Name: Contract, dtype: float64

In [41]:
# Above 80 (percentage)
total_charges_above80.Contract.value_counts() / total_charges_above80.shape[0]

Month-to-month    0.590909
One year          0.296791
Two year          0.112299
Name: Contract, dtype: float64

## 4. For those who did not churn what is the proportion of each kind of contract?

In [42]:
## Extract those who unsubscribed the services (Churn = Yes)

paying_df = pd.read_sql("select * from churn_all where Churn = 'No'", conn)

In [43]:
paying_df.TotalCharges.describe()

count    5163.000000
mean     2555.344141
std      2329.456984
min        18.800000
25%       577.825000
50%      1683.600000
75%      4264.125000
max      8672.450000
Name: TotalCharges, dtype: float64

In [44]:
# Find the 80th percentile of the data in TotalCharges
paying_df.TotalCharges.quantile(0.8)

4895.850000000001

In [45]:
# Divide the data by the 80th percentile of the data from the TotalCharges variable
paying_total_charges_under80 = paying_df.query("TotalCharges <= 4895.8")
paying_total_charges_above80 = paying_df.query("TotalCharges > 4895.8")

In [46]:
# Under 80 (percentage)
paying_total_charges_under80.Contract.value_counts() / paying_total_charges_under80.shape[0]

Month-to-month    0.500726
Two year          0.253269
One year          0.246005
Name: Contract, dtype: float64

In [47]:
# Above 80 (percentage)
paying_total_charges_above80.Contract.value_counts() / paying_total_charges_above80.shape[0]

Two year          0.572120
One year          0.280736
Month-to-month    0.147144
Name: Contract, dtype: float64

- If we want to stop Churning we would like to sell contracts to people for at least one year, and much better for 2 years (Conjecture).
- We have 57% of the people that stayed in the company in the top 20% LTV with 2 year contracts, and only 11% of the people in the top 20% had a 2 year contract and left the company.
- 95% of the people that left the company and are in the low 80% have a month-to-month contract, in contrast to only 50% of the people in the low 80% that stayed in the company. We can say that having a month-to-month contract is one of the reasons that people are leaving the company (conjecture).